### PR TITLE
simpler definition of custom charge

### PR DESCRIPTION
**Pull request by mganahl**
in `BaseCharge.__init__`, set `self.charge_types` to `[type(self)]` if `charge_types=None` is passed.
This eliminates the neccessity for the user to write their own __init__ in derived classes

### CODE REVIEW

File: tensornetwork/block_sparse/charge.py [hunk 1]
```diff
@@ -528,16 +530,6 @@ def isin(self, target_charges: Union[np.ndarray, "BaseCharge"]) -> np.ndarray:
 
 class U1Charge(BaseCharge):
   """Charge Class for the U1 symmetry group."""
-  def __init__(self,
```
Comment (Thenerdstation) [inline tensornetwork/block_sparse/charge.py:531]: Why are we removing the init function here and not for the Z2 and Modular charges?

Comment (mganahl) [reply tensornetwork/block_sparse/charge.py:531]: U1Charge doesn't perform any checks currently, the others do.

**Comment by mganahl** [comment]
let's actually wait with this. I wan't test it a bit more

File: tensornetwork/block_sparse/charge.py (hunk 1 above)
Comment (Thenerdstation) [reply tensornetwork/block_sparse/charge.py:531]: Oh I see, you're defaulting to `BaseCharge`'s init, got it.

**Comment by Thenerdstation** [comment]
Oh no missed your comment. That's ok, feel free to add more testing in the next PR

**Comment by mganahl** [comment]
np  :)

**Pull request merged**